Rainfall/Rainfall.App/RainfallCalculator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Rainfall.App
+{
+    public class RainfallCalculator : IRainfallCalculator
+    {
+        private double[] extractedRainfalls(string town, string strng)
+        {
+            List<double> rainfall = new List<double>();
+
+            List<double> emptyList = new List<double>();
+
+            string[] townDataArray = strng.Split("\n");
+
+            foreach (var townData in townDataArray)
+            {
+                string[] monthWithRainfall = townData.Split(":")[1].Split(",");
+
+                if (townData.Split(":")[0].Equals(town))
+                {
+                    foreach (var month in monthWithRainfall)
+                    {
+                        string rainfallString = month.Split(" ")[1];
+
+                        rainfall.Add(double.Parse(rainfallString, System.Globalization.CultureInfo.InvariantCulture));
+                    }
+
+                    return rainfall.ToArray();
```

Comment:
This can move out of the foreach and be returned.
The list will either have values or be null